documentation/internal/Development-Workflow.md

```diff
@@ -0,0 +1,311 @@
+
+- The file gets re-generated (by manually running the script `yarn generate`) with the complete list of all the types related to the tables, relations and fields in the GraphQL server. And adds types to manually.
+- The file also gets generated types for the `queries`, `mutation` and `fragments` defined in the next folder...
+
+#### graphql
+
+Folder storing all GraphQL queries, mutation and fragments (so everthing is in one place).
+
+**Important**: Always check that the type for the files included in this folder are using `.ts` - or generation will fail!
+
+**Rule 1**: All GraphQl calls are grouped in here to keep all-in-one-place and to be used on the the type generation.
+
+**Rule 2**: After adding, creating run the `yarn generate` script.
+
+After the generation runs a new **custom hook** with all required types is defined in the `generated/graphql.ts` file named as the pattern.
+
+Example: a new query called `getApplications.query.ts` will get a new hook (for querying) as `useGetApplicationsQuery`.
+
+**Rule 3**: Use the **Pattern**:
+
+- Query: `get` `Entity` `.` `query` `.ts`
+- Mutation: (`create` | `update`) `Entity` `.` `mutation` `.ts`
+- Fragment: (`add` | `remove`) `Entity` `.` `fragment` `.ts`
+
+##### fragments
+
+_What is a fragment?_ A partial query or mutation that can be used as part of other queries or (Mainly) for getting or setting some data to the cache. We do have to update the cache after a mutation to keep it up-to-date.
+
+##### queries
+
+_What is a query?_ Something to be fetched from the GraphQL server. Can have some input and defines what is returned.
+
+##### mutation
+
+_What is a mutation?_ Something to be posted to the GraphQL server. Habe some defined input and also something that is returned.
+
+#### helpers
+
+Add here any methods that can be used in other places - that aren't hooks or compnents itselves.
+
+#### hooks
+
+Our **custom hooks** - exaplained in more detail in the Development sections below.
+
+### Files on the root src folder
+
+- `cache.tsx` - Start the Apollo Cache.
+  Also define some local state to be used within the cache on the root level or as part of any table (@client). Read more [here](https://www.apollographql.com/docs/react/local-state/managing-state-with-field-policies/#storing-local-state-in-the-cache).
+
+- `config.json` - Some global varibles to be used during the configuration of the App. i.e: GraphQL server host.
+
+- `index.tsx` - App start - Just connects the Apollo Cache (from `cache.tsx`), add semantic-ui for UI and call the main component: `App.tsx`.
+
+---
+
+# Local vs. Remote states
+
+### Remote states
+
+The local data kept in the app by Apollo client CACHE is considered remote state since is a small mirror of the GraphQL server. The cached data is reused for up-coming queries (if we don't enforce a `refetch` from the server), just returning the local **remote state** for the queried data - when it is available. Or get from the server if not available in cache.
```

Comment:
```suggestion
The local data kept in the app by Apollo client CACHE is considered remote state since it is a small mirror of the GraphQL server. The cached data is reused for up-coming queries (if we don't enforce a `refetch` from the server), returning the local **remote state** for the queried data - when it is available. Or it fetches from the server if not available in cache.
```